### DISCUSSION TITLE
having trouble downloading pgscatalog_utils onto mac studio with  M1

### DISCUSSION THREAD

**Discussion by keri** [PGScatalog/pgsc_calc#386] [Q&A] [answered]
Has anyone else had issues with installing the pgscatalog_utils on a mac with an M1? 

I've tried several things so appreciate any feedback:

Firstly: I need to run it in a conda env as it fails to execute in local environment with Java version 23.

########## trying install in local environment ########

ERROR: Cannot find Java or it's a wrong version -- please make sure that Java 8 or later (up to 21) is installed
NOTE: Nextflow is trying to use the Java VM defined by the following environment variables:
 JAVA_CMD: /opt/homebrew/Cellar/openjdk/23/libexec/openjdk.jdk/Contents/Home/bin/java
 JAVA_HOME: 

####  I create conda env with java version 21 and it works to download: #########

% curl -s https://get.nextflow.io | bash

      N E X T F L O W
      version 24.04.4 build 5917
      created 01-08-2024 07:05 UTC (19:05 NZDT)
      cite doi:10.1038/nbt.3820
      http://nextflow.io

Nextflow installation completed. Please note:
- the executable file `nextflow` has been created in the folder: /Users/<username>/batchScripts
- you may complete the installation by moving it to a directory in your $PATH

% mv nextflow ~/bin/

#### I ensure this in my path using .zshrc file ####

export PATH=$PATH:~/bin

% source ~/.zshrc

### installed mamba in conda env. #####

% mamba install pgscatalog-util

error    libmamba Could not solve for environment specs
    The following package could not be installed
    └─ pgscatalog-utils is not installable because it requires
       └─ pgscatalog.core [>=0.2.1,<0.3.0 *|>=0.3.1,<0.4.0 *|>=0.3.2,<0.4.0], which requires
          └─ xopen >=1.8.0,<2.0.0, which does not exist (perhaps a missing channel).
critical libmamba Could not solve for environment specs

##### There are no options for xopen version >= 1.8.0 and < 2.0 on conda ####

![Screenshot 2024-10-26 at 12 08 39 PM](https://github.com/user-attachments/assets/a672b89f-481a-4e6b-88a0-7249014246dd)

**Comment by keri** [comment, marked as answer]
Just an FYI in case it helps someone else withe similar problems: 

It works with docker in the conda env (which I tried and failed at some point but was probably trying it outside of the conda env along with many other attempts). 

Here is the command that works in activated conda env:

 (conda_env)... $nextflow run pgscatalog/pgsc_calc -profile docker --input <my_samplesheet.csv>  --pgs_id PGS000020,PGS000036,PGS001294 --target_build GRCh37

**Comment by nebfield** [comment]
All of the packages are available in the bioconda or [conda-forge channels](https://anaconda.org/conda-forge/xopen). We don't use the default anaconda channel because [they have restrictive license terms](https://stackoverflow.com/a/67708768) that apply to any user downloading software from the defaults channel. It might help to make sure your channels are [set up in the right order](https://bioconda.github.io/#usage):

```
$ conda config --add channels bioconda
$ conda config --add channels conda-forge
$ conda config --set channel_priority strict
```

But you shouldn't need to install the `pgscatalog-utils` package manually. 

If you've got a conda environment where Nextflow is installed just use the `-profile conda/mamba` option. Nextflow should take care of creating each environment automatically (and check your channel order).
